src/sssom/io.py

```diff
@@ -134,34 +135,25 @@ def split_file(input_path: str, output_directory: Union[str, Path]) -> None:
 
 
 def get_metadata_and_prefix_map(
-    metadata_path: Union[None, str, Path] = None, prefix_map_mode: Optional[str] = None
-) -> Metadata:
+    metadata_path: Union[None, str, Path] = None, *, prefix_map_mode: Optional[str] = None
+) -> Tuple[Converter, MetadataType]:
     """
-    Load SSSOM metadata from a file, and then augments it with default prefixes.
+    Load SSSOM metadata from a YAML file, and then augment it with default prefixes.
 
     :param metadata_path: The metadata file in YAML format
     :param prefix_map_mode: one of metadata_only, sssom_default_only, merged
-    :return: a prefix map dictionary and a metadata object dictionary
+    :return: A converter and remaining metadata from the YAML file
     """
     if metadata_path is None:
-        return Metadata.default()
+        return get_converter(), get_default_metadata()
 
     with Path(metadata_path).resolve().open() as file:
         metadata = yaml.safe_load(file)
-    if not metadata.get("mapping_set_id"):
-        metadata["mapping_set_id"] = generate_mapping_set_id()
-    if not metadata.get("license"):
-        metadata["license"] = DEFAULT_LICENSE
-        logging.warning(f"No License provided, using {DEFAULT_LICENSE}")
```

Comment:
While your ChainMap is beautiful in terms of code readability, it does do silent repairs. I am not saying no, just pointing out that the way you have it, `sssom-py` will add certain prefixes and metadata without even informing the user that it happened.. Is there a way we can make ChainMap a bit more chatty somehow, or is it out of our control? Basically some way to report back if a piece of metadata or a prefix was added that was not previously there?

Reply:
ChainMap is implemented in C, so there's nothing we can do directly.

if desired, I can implement a chatty chain map. But, at the same time, we should reconsider why we're adding dummy values to a required field. Shouldn't we require users to for real think about this and supply something themselves? Or if we don't think so, why not just allow it to be none?

Reply:
I am trying to think of an adequate precedence. For example, if you save an ontology in protege and you don't give it an IRI, it will just add one. Maybe it is ok, for these two values. If someone complains we can reconsider this! Thanks for your thought!